### PR TITLE
Migrate S1219 from AstNode and fix SONARJS-462

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NonCaseLabelInSwitchCheck.java [hunk 1]
```diff
@@ -42,36 +38,30 @@
 @ActivatedByDefault
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("10min")
-public class NonCaseLabelInSwitchCheck extends SquidCheck<LexerlessGrammar> {
+public class NonCaseLabelInSwitchCheck extends BaseTreeVisitor {
 
   private boolean inCase;
```
Comment (benzonico) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonCaseLabelInSwitchCheck.java]: The fact that this is only a boolean is bothering me:  what about nested functions : 
```
case FOO: 
     var f = function() {
         label1: foo(); // false positive ?          
     }
```

Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonCaseLabelInSwitchCheck.java]: @benzonico indeed, I create the ticket [SONARJS-462](http://jira.sonarsource.com/browse/SONARJS-462) for the false-positive and fixed it in fbc1178d59bf622c7c6b1f3fed61e3214a92ce44

**Comment by lindamartin** [comment]
@benzonico assigned to you again to review the last commit.

**Comment by benzonico** [comment]
LGTM

**Pull request merged**